webapp/src/lib/bundler.ts

```diff
@@ -52,18 +53,35 @@ type GasLimits = {
 
 export const getUserOpGasPrice = async (
 	provider: JsonRpcProvider,
+	basePriceMultiplier?: bigint,
 ): Promise<GasFee> => {
-	const feeHistory = await provider.send("eth_feeHistory", ["0x1", "latest"]);
+	const feeHistory = await provider.send("eth_feeHistory", [
+		"0x1",
+		"latest",
+		[100],
+	]);
 	const maxPriorityFeePerGas = await provider.send(
 		"eth_maxPriorityFeePerGas",
 		[],
 	);
+	const baseFee =
+		BigInt(feeHistory.baseFeePerGas[0]) * (basePriceMultiplier ?? 2n);
 	return {
-		maxFeePerGas: feeHistory.baseFeePerGas[0],
+		maxFeePerGas: toBeHex(baseFee * BigInt(maxPriorityFeePerGas)),
```

Comment:
I recently found out about `toQuantity` which is what you want for encoding `bigint`s over JSON:

https://docs.ethers.org/v6/api/utils/#toQuantity